hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieWriteHandle.java

```diff
@@ -138,9 +139,35 @@ protected Path makeNewFilePath(String partitionPath, String fileName) {
    *
    * @param partitionPath Partition path
    */
-  protected void createMarkerFile(String partitionPath, String dataFileName) {
-    WriteMarkersFactory.get(config.getMarkersType(), hoodieTable, instantTime)
-        .create(partitionPath, dataFileName, getIOType(), config, fileId, hoodieTable.getMetaClient().getActiveTimeline());
+  protected void createInProgressMarkerFile(String partitionPath, String dataFileName, String markerInstantTime) {
+    WriteMarkers writeMarkers = WriteMarkersFactory.get(config.getMarkersType(), hoodieTable, instantTime);
+    if (!writeMarkers.doesMarkerDirExist()) {
+      throw new HoodieIOException(String.format("Marker root directory absent : %s/%s (%s)",
+          partitionPath, dataFileName, markerInstantTime));
+    }
+    if (config.enforceFinalizeWriteCheck()
+        && writeMarkers.markerExists(writeMarkers.getCompletionMarkerPath("", "FINALIZE_WRITE", markerInstantTime, IOType.CREATE))) {
```

Comment:
In general, instead of exactly match of the state, I prefer the final consistency of the files diff we have on master.

Reply:
If the job has passed through the 
(a) write stage to create the data files
(b) started a commit and have finalized writes (keeping the files that are part of the write statuses and removing the duplicate files)
(c) when updating the MDT for RLI (or before updating MDT),  if  writestatus information (RDD blocks also persisted in the containers local storage) are found to be lost due to lost/failed containers
Having this flag turned on would force the job to fail, instead of retrying the tasks/stages to recreate the data files (associated with missing write statuses).